src/executor/current_thread/mod.rs

```diff
@@ -480,7 +481,16 @@ impl<'a, P: Park> Entered<'a, P> {
     pub fn turn(&mut self, duration: Option<Duration>)
         -> Result<Turn, TurnError>
     {
-        if !self.tick() {
+        const ZERO_DURATION: Duration = Duration::from_millis(0);
+
+        let res = self.executor.park.park_timeout(ZERO_DURATION);
```

Comment:
I think that it is pretty important that `turn` only calls `park` once. Otherwise this will result in doubling syscalls (to `epoll_wait`).

Perhaps the strategy would be to always call `park` *before* trying to execute futures, but if there are futures pending processing, then the timeout is set to zero.

I think that this should satisfy the case?